Video feature

MehwishDogar opened the pull request.

MehwishDogar committed:
Adding Video-Feature:

movie-list.json:
changing the large-img to video-url and adding url of movies

App.css:
adding app-header and heading styles in it.

App.js:
adding header div and add logo img in it.

rehanumar commented on src/App.css:
```diff
@@ -1,22 +1,73 @@
+.app {
   text-align: center;
+  display: flex;
+  height: 90%;
```
@MehwishDogar changing height does not scale the design correctly. it will affect responsiveness on different devices. you can test that thing. a good alternative is to let the content decide the height of it's container (box)

rehanumar commented on src/App.css:
```diff
@@ -1,22 +1,73 @@
-.App-logo {
-  height: 40vmin;
+.app-header {
+  height: 10%;
```
same here.

rehanumar commented on src/App.css:
```diff
@@ -1,22 +1,73 @@
-  background-color: #282c34;
-  min-height: 100vh;
+.heading {
+  font-size: xx-large;
```
@MehwishDogar using a shorthand property like `font:` is considered a good practice rather then individual properties like `font-family`.
https://google.github.io/styleguide/htmlcssguide.html#Shorthand_Properties

rehanumar replied:
I still can't see all the font properties merged into one single shorthand property `font`.

rehanumar commented on src/App.css:
```diff
@@ -1,22 +1,73 @@
-  font-size: calc(10px + 2vmin);
-  color: white;
+  margin: 20px;
+  margin-top: 40px;
```
same here. use shorthand margin property to set the top `margin` rather than using `margin-top` property.

rehanumar commented on src/App.css:
```diff
@@ -1,22 +1,73 @@
+}
+
+.movie-box h4 {
+  margin: 5px 10px 5px 0px;
```
+1 for correctly using shorthand property.

rehanumar commented on src/App.css:
```diff
@@ -1,22 +1,73 @@
+  margin: 5px
+}
+
+#reset-movies {
```
good to play around with other css selectors but in one convention `ids` are generally reserved for js event binding and other DOM manipulation in old architectures of Single Page Applications (SPAs) while class selectors are used for css.

And by old architectures of Single Page Applications, I meant backbone.js, Jquery, handlebars, etc.

rehanumar replied:
@MehwishDogar this change was also not done. use class instead of id. one more problem with id is that you can't apply same id to more than one dom node. so what ever the intention is behind using `reset-movies` id. just change this to class.

rehanumar commented on src/App.css:
```diff
@@ -1,22 +1,73 @@
+
+#reset-movies {
+  margin: 0px;
+  margin-left: 5px;
```
use shorthand margin property only

rehanumar commented on src/components/FilterButton.js:
```diff
@@ -0,0 +1,18 @@
+function FilterButton (props) {
+    return (
+        <div className="filter-button">
+            <label for="free-movies">Free Movies </label>
```
this file should not show up in code review as it was previously reviewed and merged in `development` branch.

Change the PR from master to development as we want this feature to be merged in development first according to `gitflow`.
https://datasift.github.io/gitflow/IntroducingGitFlow.html

rehanumar commented on src/components/MediaPlayer.js:
```diff
@@ -0,0 +1,39 @@
+    );
+
+
+
```
to much unneccassary line breaks. use convention correctly.

rehanumar commented on src/components/MediaPlayer.js:
```diff
@@ -0,0 +1,39 @@
+import React,{ Component } from "react";
```
space after comma.

rehanumar commented on src/components/MovieBox.js:
```diff
@@ -0,0 +1,27 @@
+// TODO functional component of MovieBox which is a control component
```
remove todo.

rehanumar reviewed (changes requested):
+1 for effort on playing out with other possible options like including headers. that's the best way to learn.

some changes are mentioned.

MehwishDogar committed:
Review changes

rehanumar commented on .vscode/launch.json:
```diff
@@ -0,0 +1,15 @@
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
```
@MehwishDogar `.gitignore` everything under the `.vscode` hidden directory. and before doing git ignore. delete this launch.json file and commit/push the deleted change to remove it from here as well. as a consequence of deleting the file you might end up losing/redoing some of editor configurations mentioned in this file.

rehanumar replied:
@MehwishDogar this file is not removed. please follow the steps above.

rehanumar commented on src/App.css:
```diff
@@ -1,11 +1,30 @@
+
```
remove this starting line break.

rehanumar reviewed (changes requested):
minor changes. rest looks good.

MehwishDogar committed:
Video-feature branch

MediaPlayer.js:
Adding iframe to MediaPlayer for video fetching from youtube.

movie-list.json:
modified the video-url in movie-list.json

App.js:
adding div of app-footer.

App.css:
adding background-color in app class. adding app.footer class for footer css.

rehanumar commented on src/components/MediaPlayer.js:
```diff
@@ -1,13 +1,19 @@
-import movies from "../movie-list.json";
+import React from "react";
+import "video-react/dist/video-react.css";
+//import { Player } from 'video-react';
```
remove deprecated code.

rehanumar commented on package.json:
```diff
@@ -6,7 +6,9 @@
-    "react-scripts": "3.2.0"
+    "react-scripts": "3.2.0",
+    "redux": "^4.0.5",
+    "video-react": "^0.14.1"
```
@MehwishDogar remove deprecated library from dependencies.

rehanumar commented on src/components/MediaPlayer.js:
```diff
@@ -1,13 +1,19 @@
+            </iframe>
+         </div>
     );
+
```
extra line, I encourge to install and use prettier in this PR as well. it will resolve and will made visible a lot of styling issues.
https://github.com/prettier/prettier

rehanumar reviewed (changes requested):
minor changes

=> remove previously included library that is now deprecated after using iframes.
=> include prettier in this PR to resolve styling issues in this PR and in future PR as well.

kudos for effort and for being in growth mindset.

MehwishDogar committed:
changes done
